sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/AvroReaderSyncFactory.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.internal.avro.implementation;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Factory class for {@link AvroSyncReader}.
+ */
+public class AvroReaderSyncFactory {
+
+    /**
+     * Gets a new instance of {@link AvroSyncReader} with support for offset and thresholdIndex.
+     *
+     * @param data The ByteBuffer that contains the Avro data.
+     * @param offset The position in the ByteBuffer from where to start parsing.
+     * @param thresholdIndex The minimum index of the objects to be returned.
+     * @return An AvroSyncReader.
+     */
+    public AvroSyncReader getAvroReader(ByteBuffer data, int offset, long thresholdIndex) {
+        return new AvroSyncReader() {
+            private final AvroSyncParser parser = new AvroSyncParser(true); // assuming true means handling partial reads
```

Comment:
This implementation is a bit different than the asynchronous one for reading Avro with a header and a body